flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/CharVarCharTrimPadCastRule.java

```diff
@@ -43,19 +43,18 @@
  * the one specified by the precision of the target {@link LogicalTypeRoot#CHAR} or {@link
  * LogicalTypeRoot#VARCHAR} type.
  */
-class CharacterFamilyTrimmingAndPaddingCastRule
-        extends AbstractNullAwareCodeGeneratorCastRule<StringData, StringData> {
+class CharVarCharTrimPadCastRule
+        extends AbstractNullAwareCodeGeneratorCastRule<Object, StringData> {
 
-    static final CharacterFamilyTrimmingAndPaddingCastRule INSTANCE =
-            new CharacterFamilyTrimmingAndPaddingCastRule();
+    static final CharVarCharTrimPadCastRule INSTANCE = new CharVarCharTrimPadCastRule();
 
-    private CharacterFamilyTrimmingAndPaddingCastRule() {
+    private CharVarCharTrimPadCastRule() {
         super(
                 CastRulePredicate.builder()
                         .predicate(
                                 (inputType, targetType) ->
                                         targetType.is(LogicalTypeFamily.CHARACTER_STRING)
-                                                && !isStringType(targetType))
+                                                && !targetType.equals(STRING_TYPE))
```

Comment:
can this also be converted to a non-predicate now?

Reply:
Nope, because this is a `not equals`.